src/Groups/GAPGroups.jl

```diff
@@ -612,7 +661,7 @@ Return the vector of all conjugacy classes of subgroups of G.
 # Examples
 ```jldoctest
 julia> G = symmetric_group(3)
-Sym( [ 1 .. 3 ] )
+Permutation group of degree 3 and order 6
```

Comment:
It would be kinda nice to preserve GAP assigned names to some extent; in this case printing this group as `Sym(3)` or `Symmetric group of degree 3` or anything like would be nice

Reply:
Of course we can also take inspiration from Magma and Sage and GAP...

Magma:
```
> Sym(6);
Symmetric group acting on a set of cardinality 6
Order = 720 = 2^4 * 3^2 * 5
```
so a special name, and the order is also printed factorized, which I guess can be quite handy.
Or here:
```
> H, map := PermutationGroup< 9 | (1,2,4)(5,6,8)(3,9,7), (4,5,6)(7,9,8) >;
> H;
Permutation group H acting on a set of cardinality 9
    (1, 2, 4)(3, 9, 7)(5, 6, 8)
    (4, 5, 6)(7, 9, 8)
```
So the generators are still printed in this detailed form... But they are hidden in "compact" form, e.g. when printing the map:
```
> map;
Mapping from: GrpPerm: H to GrpPerm: $, Degree 9, Order 2^7 * 3^4 * 5 * 7
```
Note the use of `H` and `$` which are the "names" of the group in the session. We can see this when we assign a name to the codomain
```
> Codomain(map);
Symmetric group acting on a set of cardinality 9
Order = 362880 = 2^7 * 3^4 * 5 * 7
> U := Codomain(map);
> map;
Mapping from: GrpPerm: H to GrpPerm: U
```
(In our system, assigning a name would drop the `GrpPerm` / `Permutation group` bit)


(Should look at how they handle pc and fp groups later, too; and perhaps also look at Sage... well, and GAP, but I know that / that's basically what we are using right now)

Reply:
I would distinguish assigning a name to the group from assigning the group object to a variable. Why shall the name of the first variable to which the group gets assigned determine how the group is printed?
(Yes, this happens already for certain Oscar objects, but I find this unnatural.)

Concerning the idea to assign a name to a group and then use this name for printing: This is in principle reasonable. However, I find it irritating that calling `describe` (or rather GAP`s `StructureDescription`) *automatically* assigns a name, as in the following GAP session.
```
gap> l:= [ Group( (1,2) ), SmallGroup( 2, 1 ), Group( [[Z(3)]] ) ];
[ Group([ (1,2) ]), <pc group of size 2 with 1 generator>, 
  Group([ [ [ Z(3) ] ] ]) ]
gap> List( l, StructureDescription );
[ "C2", "C2", "C2" ]
gap> l;
[ C2, C2, C2 ]
```
(The same happens in `all_primitive_groups`.)
Thus the user does not have control over the assigned names.